File: InventoryManagement.Core/Entities/Item.cs
using InventoryManagement.Core.Common;

namespace InventoryManagement.Core.Entities;

public class Item : BaseEntity
{
    public string ItemCode { get; set; } = string.Empty;
    public string ItemName { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int CategoryId { get; set; }
    public int UnitOfMeasureId { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal? ReorderLevel { get; set; }
    public string? ImageUrl { get; set; }

    // Navigation Properties
    public Category Category { get; set; } = null!;
    public UnitOfMeasure UnitOfMeasure { get; set; } = null!;
    public ItemStock? ItemStock { get; set; }
    public ICollection<PurchaseOrderItem> PurchaseOrderItems { get; set; } = new List<PurchaseOrderItem>();
    public ICollection<SalesOrderItem> SalesOrderItems { get; set; } = new List<SalesOrderItem>();
    public ICollection<QuotationItem> QuotationItems { get; set; } = new List<QuotationItem>();
}

File: InventoryManagement.UI/src/app/core/models/item.ts
export interface Item {
  id: number;
  itemCode: string;
  itemName: string;
  categoryId: number;
  categoryName: string;
  unitOfMeasureId: number;
  unitOfMeasureName: string;
  unitPrice: number;
  reorderLevel?: number;
  imageUrl?: string;
  quantityOnHand: number;
  quantityAvailable: number;
  createdAt: Date;
}

export interface CreateItemRequest {
  itemCode: string;
  itemName: string;
  description?: string;
  categoryId: number;
  unitOfMeasureId: number;
  unitPrice: number;
  reorderLevel?: number;
  imageUrl?: string;
}

export interface UpdateItemRequest {
  itemCode: string;
  itemName: string;
  description?: string;
  categoryId: number;
  unitOfMeasureId: number;
  unitPrice: number;
  reorderLevel?: number;
  imageUrl?: string;
}